aggregator/src/aggregator.rs

```diff
@@ -1883,6 +1883,33 @@ impl VdafOps {
             }
         };
 
+        // Check if any two extensions have the same extension type across public
+        // and private extension fields. If so, the Aggregator MUST mark the input
+        // share as invalid with error invalid_message. (§4.6.2.4 step 7)
+        //
+        // Note at this point we can't provide Error::InvalidMessage here because
+        // we're packaging this into a report rejection list.
+        let mut extensions = HashMap::new();
+        if !leader_private_extensions
```

Comment:
I don't suppose that the existing code that handles this on the helper side could be factored out into a method used by both? Then again it's so little code that it may not be worth it.

Reply:
Not really. The inputs are different, and so are the metrics. The only part that is generic is this:
```
.all(|extension| {
                                extensions
                                    .insert(*extension.extension_type(), extension.extension_data())
                                    .is_none()
                            }) {
                                debug!(
                                    task_id = %task.id(),
                                    report_id = ?prepare_init.report_share().metadata().id(),
                                    "Received report share with duplicate extensions",
                                );
```

which isn't nothin', but basically just the lambda inside `all` is actually generic.